lib/tree.rb

```diff
@@ -1,38 +1,58 @@
 class NoApplesError < StandardError; end
 
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @height = 0
+    @age = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+
+    @alive = false if @age > 100
+    rand(176..882).times {add_apples} if @age >= 5
+    # Height approaches 30 feet
+    @height = 30 - (30 / (age + 1.0)).round(2)
   end
 
   def add_apples
+    @apples << Apple.new('red', rand(2..4))
   end
 
   def any_apples?
+    @apples.count > 0
   end
 
   def pick_an_apple!
-    raise NoApplesError, "This tree has no oranges" unless self.any_apples?
+    raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    @apples.pop
   end
 
   def dead?
+    not @alive
```

Comment:
There's a difference with regards to precedence and implementation. However, in this instance it was a readability decision. That said, I'm not tied to `not`, and could be persuaded to use `!` instead.